### PR TITLE
Clarify what a "network error" is

### PR DESCRIPTION
**Pull request by skriptble**
I think adding a term should be enough to clear up the ambiguity.

### CODE REVIEW

File: source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst [hunk 1]
```diff
@@ -200,6 +200,11 @@ address
 
 The hostname or IP address, and port number, of a MongoDB server.
 
+network error
+`````````````
+
+An error that occurs while reading from or writing to a network socket.
```
Comment (xdg) [inline source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst]: Do you want to clarify that this is "excluding a timeout"?

Comment (skriptble) [reply source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst]: We could add it but at least in the section where the ambiguity exists, network errors and network timeouts are separate, at least that's how I read it.

> If there is a *network error or timeout* on the connection before the handshake completes

> If there is a *network timeout* on the connection after the handshake completes

Then again, a following sentence refers to a network error as "other network error"

> If, however, there is some *other network error* on the connection after the handshake completes

But I can add that clarification if we think it'll help.

Comment (xdg) [reply source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst]: Maybe leave this definition as is and add a definition for "network timeout" as being either a network error indicating a socket timeout like ETIMEDOUT or a user-space timeout like with `select`?

Comment (skriptble) [reply source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst]: Would "A timeout that occurs while reading from or writing to a network socket" work? It sounds like we need to have a larger discussion around what a "network timeout" is. Not sure if that's in scope for this change though.

Comment (skriptble) [reply source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst]: Added a definition for network timeout.

**Pull request merged**